opening-hours/src/localization/coordinates.rs

```diff
@@ -1,28 +1,23 @@
-use chrono::{Datelike, NaiveDate};
+use chrono::NaiveDate;
 use opening_hours_syntax::rules::time::TimeEvent;
-use sunrise_next::{DawnType, SolarDay, SolarEvent};
+use sunrise::{DawnType, SolarDay, SolarEvent};
 
 /// A valid pair of geographic coordinates.
 ///
 /// See https://en.wikipedia.org/wiki/Geographic_coordinate_system
 #[derive(Clone, Copy, Debug, PartialEq, PartialOrd)]
-pub struct Coordinates {
-    lat: f64,
-    lon: f64,
-}
+pub struct Coordinates(sunrise::Coordinates);
 
 impl Coordinates {
     /// Validate a pair of latitude / longitude.
     ///
     /// Return `None` if values are out of range (`abs(lat) > 90` or
     /// `abs(lon) > 180`).
     pub const fn new(lat: f64, lon: f64) -> Option<Self> {
-        if lat.is_nan() || lon.is_nan() || lat < -90.0 || lat > 90.0 || lon < -180.0 || lon > 180.0
-        {
-            return None;
+        match sunrise::Coordinates::new(lat, lon) {
+            Some(c) => Some(Self(c)),
+            None => None,
         }
```

Comment:
```suggestion
        sunrise::Coordinates::new(lat, lon).map(Self)
```

Reply:
Nice to learn, thanks 🤯

Reply:
Yea, there isn't a way to express that a function would be `const` if the function passed in (in this case `Self`) is also `const`. Maybe we'll get that some day :/